README.md
# Directa 24 Java client library


The official [Directa 24][directa24] Java client library.

## Installation

### Requirements

- Java 1.8 or later

### Gradle users

Add this dependency to your project's build file:

```groovy
implementation "com.directa24:cashin-java-sdk:1.0.11"
```

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>com.directa24</groupId>
  <artifactId>cashin-java-sdk</artifactId>
  <version>1.0.11</version>
</dependency>
```

## Documentation

Please see the [Java API docs][api-docs] for the most up-to-date documentation.

## Usage

#### Deposit Credentials

```java
String depositKeySbx = "1955f2d";
String secretKeySbx = "4lph0ns3";

Directa24 directa24Sandbox = new Directa24.Sandbox(depositKeySbx, secretKeySbx);
```

#### Read Only Credentials

```java
String readOnlyKeySbx = "EgIECaVuyW";

Directa24 directa24Sandbox = new Directa24.Sandbox(readOnlyKeySbx);
```

#### Create Deposit

```java
public class CreateDepositExample {

   public static void main(String[] args) {

      Address address = Address.builder()
                               .street("Rua Dr. Franco Ribeiro, 52")
                               .city("Rio Branco")
                               .state("AC")
                               .zipCode("11600-234")
                               .build();

      Payer payer = Payer
            .builder()
            .id("4-9934519")
            .address(address)
            .document("21329039050")
            .documentType("CPF")
            .email("juanCarlos@hotmail.com")
            .firstName("Ricardo")
            .lastName("Carlos")
            .phone("+598 99730878")
            .build();

      BankAccount bankAccount = BankAccount
            .builder()
            .bankCode("01")
            .accountNumber("3242342")
            .accountType("SAVING")
            .beneficiary("Ricardo Carlos")
            .branch("12")
            .build();

      CreateDepositRequest createDepositRequest = CreateDepositRequest
            .builder()
            .invoiceId("108")
            .amount(new BigDecimal(100))
            .country("BR")
            .currency("BRL")
            .payer(payer)
            .paymentMethod("BB")
            .paymentType("BANK_TRANSFER")
            .bankAccount(bankAccount)
            .earlyRelease(false)
            .feeOnPayer(false)
            .surchargeOnPayer(false)
            .bonusAmount(BigDecimal.ONE)
            .bonusRelative(false)
            .strikethroughPrice(BigDecimal.ONE)
            .description("Test")
            .clientIp("186.51.171.84")
            .language("es")
            .deviceId("00000000-00000000-01234567-89ABCDEF")
            .backUrl("https://yoursite.com/deposit/108/cancel")
            .successUrl("https://yoursite.com/deposit/108/confirm")
            .errorUrl("https://yoursite.com/deposit/108/error")
            .notificationUrl("https://yoursite.com/ipn")
            .logo("https://yoursite.com/logo.png")
            .test(true)
            .mobile(false)
            .idempotency("")
            .build();

      try {
         CreateDepositResponse createDepositResponse = directa24Sandbox.client.createDeposit(createDepositRequest);
         
         // Handle response

      } catch (Directa24Exception e) {
         // Handle errors
      }
   }
}
```
#### Deposit Status

```java
public class DepositStatusExample {

   public static void main(String[] args) {
      DepositStatusRequest depositStatusRequest = DepositStatusRequest
                                                  .builder()
                                                  .id(300000001)
                                                  .build();
      try {
         DepositStatusResponse depositStatusResponse = directa24Sandbox.client.depositStatus(depositStatusRequest);
      
         // Handle response

      } catch (Directa24Exception e) {
         // Handle errors
      }
   }
}
```

#### Payment Methods

```java
public class PaymentMethodsExample {

   public static void main(String[] args) {
      PaymentMethodRequest paymentMethodRequest = PaymentMethodRequest
                                                  .builder()
                                                  .country("BR")
                                                  .build();
      try {
         List<PaymentMethodResponse> paymentMethodResponse = directa24Sandbox.client.paymentMethods(paymentMethodRequest);
      
         // Handle response

      } catch (Directa24Exception e) {
         // Handle errors
      }
   }
}
```

#### Exchange rates

```java
public class ExchangeRateExample {

   public static void main(String[] args) {
      ExchangeRateRequest exchangeRatesRequest = ExchangeRateRequest
                                                 .builder()
                                                 .country("BR")
                                                 .amount(BigDecimal.TEN)
                                                 .build();
      try {
         ExchangeRateResponse exchangeRateResponse = directa24Test.client.exchangeRates(exchangeRateRequest);
         
         // Handle response

      } catch (Directa24Exception e) {
         // Handle errors
      }
   }
}
```
For more examples see the bundled tests.

### Dependencies

The library uses [Project Lombok][lombok]. While it is not a requirement, you
might want to install a [plugin][lombok-plugins] for your favorite IDE to
facilitate development.

[JUnit 4][junit] and [Wiremock][wiremock] library are needed to run the bundled tests.

[directa24]: https://directa24.com
[api-docs]: https://docs.directa24.com/deposits-api
[lombok]: https://projectlombok.org
[lombok-plugins]: https://projectlombok.org/setup/overview
[junit]: https://junit.org/junit4/
[wiremock]: http://wiremock.org 

CHANGELOG.md
# Changelog

## 1.0.11 - 2020-09-04
* Minor fixes

## 1.0.10 - 2020-09-04
* Support for read only key
* Remove api key

## 1.0.9 - 2020-09-01
* Remove unused fields on refund creation service

## 1.0.8 - 2020-08-21
* Create refund and refund status services
* Banks service

## 1.0.7 - 2020-08-10
* Remove ending "/" from create deposits path
* Various dependencies versions update

## 1.0.4 - 2020-07-27
* Ignore unrecognized fields on deserialization
* Add iframe field to create deposit response

## 1.0.1 - 2020-07-16
* Rename deposits and exchange_rates endpoint

## 1.0.0 - 2020-07-01
* First version of the library

src/main/java/com/directa24/Main.java
package com.directa24;

import com.directa24.exception.Directa24Exception;
import com.directa24.model.request.DepositStatusRequest;
import com.directa24.model.response.DepositStatusResponse;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

public class Main {

   public static void main(String[] args) {
      String depositKeySbx = "1955f2d";
      String apiKeySbx = "hola";
      String secretKeySbx = "4lph0ns3";

      Directa24 directa24Sandbox = new Directa24.Sandbox(depositKeySbx, apiKeySbx, secretKeySbx, 30, 30);

      //      String depositKeyProd = "loginProd";
      //      String apiKeyProd = "hola";
      //      String secretKeyProd = "secretKeyProd";
      //
      //      Directa24 directa24Production = new Directa24.Production(depositKeyProd, apiKeyProd, secretKeyProd);

      DepositStatusRequest depositStatusRequest = DepositStatusRequest.builder() //
                                                                      .id(300000001) //
                                                                      .build();

      try {
         DepositStatusResponse depositStatusResponse = directa24Sandbox.client.depositStatus(depositStatusRequest);
         System.out.println(new ObjectMapper().writeValueAsString(depositStatusResponse));
      } catch (Directa24Exception | JsonProcessingException e) {
         e.printStackTrace();
      }

      //      try {
      //
      //         Address address = Address.builder()
      //                                  .street("Calle 13")
      //                                  .city("Montevideo")
      //                                  .state("AC")
      //                                  .zipCode("11600-234")
      //                                  .build();
      //
      //         Payer payer = Payer.builder()
      //                            .id("4-9934519")
      //                            .address(address)
      //                            .document("21329039050")
      //                            .documentType("CPF")
      //                            .email("juanCarlos@hotmail.com")
      //                            .firstName("Ricardo")
      //                            .lastName("Carlos")
      //                            .phone("+598 99730878")
      //                            .build();
      //
      //         CreateDepositRequest createDepositRequest = CreateDepositRequest.builder()
      //                                                                         .invoiceId("100")
      //                                                                         .amount(new BigDecimal(100))
      //                                                                         .country("BR")
      //                                                                         .currency("BRL")
      //                                                                         .payer(payer)
      //                                                                         .build();
      //
      //         CreateDepositResponse createDepositResponse = directa24Sandbox.client.createDeposit(createDepositRequest);
      //         System.out.println(new ObjectMapper().writeValueAsString(createDepositResponse));
      //      } catch (Directa24Exception | JsonProcessingException e) {
      //         e.printStackTrace();
      //      }

      //      try {
      //
      //         Address address = Address.builder() //
      //                                  .street("Calle 13") //
      //                                  .city("Montevideo") //
      //                                  .state("AC") //
      //                                  .zipCode("11600-234") //
      //                                  .build();
      //
      //         Payer payer = Payer
      //               .builder()
      //               .id("4-9934519")
      //               .address(address)
      //               .document("21329039050")
      //               .documentType("CPF")
      //               .email("juanCarlos@hotmail.com")
      //               .firstName("Ricardo")
      //               .lastName("Carlos")
      //               .phone("+598 99730878")
      //               .build();
      //
      //         BankAccount bankAccount = BankAccount
      //               .builder()
      //               .bankCode("01")
      //               .accountNumber("3242342")
      //               .accountType("SAVING")
      //               .beneficiary("Ricardo Carlos")
      //               .branch("12")
      //               .build();
      //
      //         CreateDepositRequest createDepositRequest = CreateDepositRequest
      //               .builder()
      //               .invoiceId("108")
      //               .amount(new BigDecimal(100))
      //               .country("BR")
      //               .currency("BRL")
      //               .payer(payer)
      //               .paymentMethod("BB")
      //               .paymentType("BANK_TRANSFER")
      //               .bankAccount(bankAccount)
      //               .earlyRelease(false)
      //               .feeOnPayer(false)
      //               .surchargeOnPayer(false)
      //               .bonusAmount(BigDecimal.ONE)
      //               .bonusRelative(false)
      //               .strikethroughPrice(BigDecimal.ONE)
      //               .description("Test")
      //               .clientIp("186.51.171.84")
      //               .language("es")
      //               .deviceId("00000000-00000000-01234567-89ABCDEF")
      //               .backUrl("")
      //               .successUrl("")
      //               .errorUrl("")
      //               .notificationUrl("")
      //               .logo("")
      //               .test(true)
      //               .mobile(false)
      //               .idempotency("")
      //               .build();
      //
      //         CreateDepositResponse createDepositResponse = directa24Sandbox.client.createDeposit(createDepositRequest);
      //         System.out.println(new ObjectMapper().writeValueAsString(createDepositResponse));
      //      } catch (Directa24Exception | JsonProcessingException e) {
      //         e.printStackTrace();
      //      }

      //      PaymentMethodRequest paymentMethodRequest = PaymentMethodRequest.builder() //
      //                                                                      .country("BR") //
      //                                                                      .build();
      //
      //      try {
      //         List<PaymentMethodResponse> paymentMethodResponse = directa24Sandbox.client.paymentMethods(paymentMethodRequest);
      //         System.out.println(new ObjectMapper().writeValueAsString(paymentMethodResponse));
      //      } catch (Directa24Exception | JsonProcessingException e) {
      //         e.printStackTrace();
      //      }

   }

}
